runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/fn/control/ProcessRemoteBundleOperation.java

```diff
@@ -84,10 +192,20 @@ public void start() throws Exception {
 
   @Override
   public void process(Object inputElement) throws Exception {
-    LOG.debug(String.format("Sending value: %s", inputElement));
+    LOG.error("Sending element: {}", inputElement);
+    String mainInputPCollectionId = executableStage.getInputPCollection().getId();
+    FnDataReceiver<WindowedValue<?>> mainInputReceiver =
+        remoteBundle.getInputReceivers().get(mainInputPCollectionId);
+
     try (Closeable scope = context.enterProcess()) {
-      Iterables.getOnlyElement(remoteBundle.getInputReceivers().values())
-          .accept((WindowedValue<InputT>) inputElement);
+      mainInputReceiver.accept((WindowedValue<InputT>) inputElement);
+    } catch (Exception e) {
+      LOG.error(
```

Comment:
Why survive this? It seemed actually quite important to fail.

Reply:
Now throws RuntimeException